### PR TITLE
feat: custom 404 message when there is a lack of permission

### LINKED ISSUE #26: Show 404 content when unauthorized or page doesn't exist
We should show 404 as the page content if the user types a wrong URL of if he types a correct URL but doesn't have the correct role.

image

image

Also don't forget the title to change as well

### PR DESCRIPTION
**Pull request by CuriouzK0d3r**
resolves #26

### CODE REVIEW

File: apps/artemis-web/lib/helpers.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import jwt from 'jsonwebtoken';
+import Head from 'next/head';
```
Comment (slowr) [inline apps/artemis-web/lib/helpers.ts:2]: unused import

File: apps/artemis-web/tsconfig.json [hunk 2]
```diff
@@ -12,6 +12,6 @@
     "resolveJsonModule": true,
     "isolatedModules": true
   },
-  "include": ["**/*.ts", "**/*.tsx", "next-env.d.ts"],
+  "include": ["**/*.ts", "**/*.tsx", "next-env.d.ts", "pages/404.js"],
```
Comment (slowr) [inline apps/artemis-web/tsconfig.json:15]: ?

**Review by slowr — CHANGES_REQUESTED** [review]
LGTM. Just change the error code for permission which is 403

File: apps/artemis-web/lib/helpers.ts (hunk 1 above)
Comment (CuriouzK0d3r) [reply apps/artemis-web/lib/helpers.ts]: removed

File: apps/artemis-web/tsconfig.json (hunk 2 above)
Comment (CuriouzK0d3r) [reply apps/artemis-web/tsconfig.json]: the IDE inserted that. removed now

File: apps/artemis-web/pages/adminpanel.tsx [hunk 3]
```diff
@@ -2,13 +2,28 @@ import Head from 'next/head';
...
+import DefaultErrorPage from 'next/error';
 
 const AdminPanelPage = (props) => {
   const [user, loading] = useJWT();
 
   const router = useRouter();
   if (!user && !loading) router.push('signin');
 
+  if (user && user.role !== 'admin') {
+    return (
```
Comment (slowr) [inline apps/artemis-web/pages/adminpanel.tsx:14]: try to move this part to an external function to remove duplication

Comment (CuriouzK0d3r) [reply apps/artemis-web/pages/adminpanel.tsx]: I added a hoc to solve this.

**Review by slowr — APPROVED** [review]
LGTM

**Pull request merged**